CMakeLists.txt

```diff
@@ -98,6 +98,11 @@ endif()
 
 option(ZSTD_FROM_INTERNET "Download and use libzstd from the Internet" ${ZSTD_FROM_INTERNET_DEFAULT})
 find_package(zstd 1.1.2 REQUIRED)
+if(TARGET zstd::libzstd_shared)
```

Comment:
Please don't do this. Modify the `ZSTD::ZSTD` alias library in https://github.com/ccache/ccache/blob/master/cmake/Findzstd.cmake to point to the `zstd::libzstd_shared`.

There is no need to make CMake code uglier than it has to be.

Reply:
`find_package(zstd 1.1.2 REQUIRED)` needs to change to `find_package(zstd 1.1.2 MODULE REQUIRED)`.

Then in https://github.com/ccache/ccache/blob/master/cmake/Findzstd.cmake you change:

```cmake
...
else()
  find_package(zstd CONFIG)
  if (zstd_FOUND and TARGET zstd::libzstd_shared)
    add_library(ZSTD::ZSTD ALIAS zstd::libzstd_shared)
  else()
    find_library(ZSTD_LIBRARY zstd)
    find_path(ZSTD_INCLUDE_DIR zstd.h)

    include(FindPackageHandleStandardArgs)
    find_package_handle_standard_args(
      zstd "please install libzstd or use -DZSTD_FROM_INTERNET=ON"
      ZSTD_INCLUDE_DIR ZSTD_LIBRARY)
    mark_as_advanced(ZSTD_INCLUDE_DIR ZSTD_LIBRARY)

    add_library(ZSTD::ZSTD UNKNOWN IMPORTED)
    set_target_properties(
      ZSTD::ZSTD
      PROPERTIES
      IMPORTED_LOCATION "${ZSTD_LIBRARY}"
      INTERFACE_INCLUDE_DIRECTORIES "${ZSTD_INCLUDE_DIR}")
  endif()
endif()
```